jovo-integrations/jovo-db-postgresql/src/PostgreSQL.ts

```diff
@@ -0,0 +1,170 @@
+import _merge = require('lodash.merge');
+import { ConnectionConfig, Pool } from 'pg';
+
+import { BaseApp, Db, ErrorCode, Jovo, JovoError, Log, PluginConfig } from 'jovo-core';
+
+export interface Config extends PluginConfig {
+  tableName?: string;
+  primaryKeyColumn?: string;
+  dataColumnName?: string;
+  connection?: ConnectionConfig;
+}
+
+export class PostgreSQL implements Db {
+  config: Config = {
+    dataColumnName: 'userData',
+    primaryKeyColumn: 'userId',
+    tableName: 'users',
+  };
+  /**
+   * We use single queries using the pool object as it's the recommended way of doing things
+   * when you only need a small number of queries. In our case that's two queries
+   * per request-response cycle.
+   * @see https://node-postgres.com/features/pooling#single-query
+   */
+  pool!: Pool; // initialized in install() not constructor
+
+  needsWriteFileAccess = false;
+
+  constructor(config?: Config) {
+    if (config) {
+      this.config = _merge(this.config, config);
+    }
+  }
+
+  install(app: BaseApp) {
+    if (!this.config.connection) {
+      throw new JovoError(
+        'A connection config is needed.',
+        ErrorCode.ERR_PLUGIN,
+        'jovo-db-postgresql',
+        'PostgreSQL connection configuration has to be set correctly.',
+        'https://www.jovo.tech/marketplace/jovo-db-postgresql#installation',
+      );
+    }
+
+    this.pool = new Pool(this.config.connection);
+
+    this.setAsActiveDbIntegration(app);
+  }
+
+  async uninstall(app: BaseApp) {
+    if (this.pool) {
+      await this.pool.end();
```

Comment:
We could also listen to the exit event. However, only synchronous calls would be possble.
```
app.on('exit', (jovo) => {
      callback(jovo);
    });
```